src/benchmark_api_internal.h

```diff
@@ -41,6 +41,10 @@ class BenchmarkInstance {
   int threads() const { return threads_; }
   void Setup() const;
   void Teardown() const;
+  std::unique_ptr<ThreadRunnerBase> GetUserThreadRunner() const {
+    return benchmark_.threadrunner_ ? benchmark_.threadrunner_(threads_)
+                                    : nullptr;
+  }
```

Comment:
I think it would be better if `GetUserThreadRunner` would return the factory,
and we'd never to have to second-guess it's output later on,
and explcitly call it with the thread count, once.
```
  ??? GetThreadRunner() const {
    return benchmark_.threadrunner_ ? benchmark_.threadrunner_
                                    : [](int n){ return std::make_unique<ThreadRunnerDefault>(n); };
  }
```